avogadro/qtplugins/noncovalent/noncovalent.cpp

```diff
@@ -310,69 +320,84 @@ QWidget *NonCovalent::setupWidget()
-  distance_spin->setValue(m_angleToleranceDegrees);
-  QObject::connect(distance_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &NonCovalent::setMaximumDistance);
+  for (Index i = 0; i < 3; i++) {
+  	// angle tolerance
+		QDoubleSpinBox *angle_spin = new QDoubleSpinBox;
+		angle_spin->setRange(0.0, 180.0);
+		angle_spin->setSingleStep(1.0);
+		angle_spin->setDecimals(0);
+		angle_spin->setSuffix(tr(" °"));
+		angle_spin->setValue(m_angleTolerancesDegrees[i]);
+		QObject::connect(angle_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this,
+			[this, i](float width){ return setAngleTolerance(width, i); }
+		);
+		
+		// maximum distance
+		QDoubleSpinBox *distance_spin = new QDoubleSpinBox;
+		distance_spin->setRange(1.0, 10.0);
+		distance_spin->setSingleStep(0.1);
+		distance_spin->setDecimals(1);
+		distance_spin->setSuffix(tr(" Å"));
+		distance_spin->setValue(m_maximumDistances[i]);
+		QObject::connect(distance_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, 
+			[this, i](float width){ return setMaximumDistance(width, i); }
+		);
   
-  // line width
-  QDoubleSpinBox* lineWidth_spin = new QDoubleSpinBox;
-  lineWidth_spin->setRange(1.0, 10.0);
-  lineWidth_spin->setSingleStep(0.5);
-  lineWidth_spin->setDecimals(1);
-  lineWidth_spin->setValue(m_lineWidths[0]);
-  QObject::connect(lineWidth_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &NonCovalent::setLineWidth);
-
-  QFormLayout *form = new QFormLayout;
-  form->addRow(QObject::tr("Angle tolerance:"), angle_spin);
-  form->addRow(QObject::tr("Maximum distance:"), distance_spin);
-  form->addRow(QObject::tr("Line width:"), lineWidth_spin);
-  v->addLayout(form);
-
+  	// line width
+		QDoubleSpinBox* lineWidth_spin = new QDoubleSpinBox;
+		lineWidth_spin->setRange(1.0, 10.0);
+		lineWidth_spin->setSingleStep(0.5);
+		lineWidth_spin->setDecimals(1);
+		lineWidth_spin->setValue(m_lineWidth);
+		QObject::connect(lineWidth_spin, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &NonCovalent::setLineWidth);
+		
+		QFormLayout *form = new QFormLayout;
+  	form->addRow(QObject::tr("Angle tolerance:"), angle_spin);
+  	form->addRow(QObject::tr("Maximum distance:"), distance_spin);
+  	form->addRow(QObject::tr("Line width:"), lineWidth_spin);
+  	
+  	QWidget *page = new QWidget;
+  	page->setLayout(form);
+  	tabs->addTab(page, INTERACTION_NAMES[i]);
+  }
+  
+  v->addWidget(tabs);
   v->addStretch(1);
   widget->setLayout(v);
   return widget;
 }
 
-void NonCovalent::setAngleTolerance(float angleTolerance)
+void NonCovalent::setAngleTolerance(float angleTolerance, Index index)
 {
-  m_angleToleranceDegrees = float(angleTolerance);
+  m_angleTolerancesDegrees[index] = float(angleTolerance);
   emit drawablesChanged();
 
   QSettings settings;
-  settings.setValue("nonCovalent/angleTolerance", m_angleToleranceDegrees);
+  settings.setValue(QString((
+  	std::ostringstream() << "nonCovalent/angleTolerance" << index
```

Comment:
It looks like you want to add a number after the `nonCovalent/angleTolerance` part.

Use:
`QString("nonCovalent/angleTolerance%1").arg(index)`

https://doc.qt.io/qt-5/qstring.html#arg-5

Reply:
Wow, Qt really offers such nice functionality...